mainServiceController/src/main/java/org/venice/piazza/servicecontroller/async/AsynchronousServiceWorker.java

```diff
@@ -0,0 +1,342 @@
+					exception.getMessage(), instance.getServiceId(), instance.getInstanceId(), instance.getJobId(),
+					instance.getNumberErrorResponses()), PiazzaLogger.WARNING);
+		}
+	}
+
+	/**
+	 * Handles a non-success completed Status message.
+	 * 
+	 * @param instance
+	 *            The service instance.
+	 * @param serviceStatus
+	 *            The StatusUpdate received from the external User Service
+	 */
+	private void processErrorStatus(String jobId, String status, String message) {
+		// Remove the Instance from the Instance Table
+		accessor.deleteAsyncServiceInstance(jobId);
+
+		// Create a new Status Update to send to the Job Manager.
+		StatusUpdate statusUpdate = new StatusUpdate();
+		statusUpdate.setStatus(status);
+		// Create the Message for the Error Result of the Status
+		ErrorResult errorResult = new ErrorResult();
+		errorResult.setMessage(message);
+		statusUpdate.setResult(errorResult);
+
+		// Send the Job Status through Kafka.
+		try {
+			ProducerRecord<String, String> prodRecord = new ProducerRecord<String, String>(
+					String.format("%s-%s", JobMessageFactory.UPDATE_JOB_TOPIC_NAME, SPACE), jobId,
+					objectMapper.writeValueAsString(statusUpdate));
+			producer.send(prodRecord);
+		} catch (JsonProcessingException exception) {
+			// The message could not be serialized. Record this.
+			exception.printStackTrace();
+			logger.log("Could not send Error Status to Job Manager. Error serializing Status: " + exception.getMessage(),
+					PiazzaLogger.ERROR);
+		}
+	}
+
+	/**
+	 * Sends the cancellation status to the external User Service for the specified instance.
+	 * 
+	 * @param instance
+	 *            The instance to be cancelled
+	 */
+	@Async
+	public void sendCancellationStatus(AsyncServiceInstance instance) {
+		// Send the DELETE request to the external User Service
+		Service service = accessor.getServiceById(instance.getServiceId());
+		String url = String.format("%s/%s/%s", service.getUrl(), DELETE_ENDPOINT, instance.getInstanceId());
+		try {
+			restTemplate.delete(url);
+		} catch (HttpClientErrorException | HttpServerErrorException exception) {
+			// The cancellation sent back an error. Log it.
+			logger.log(String.format(
+					"Error Cancelling Service Instance on external User Service: HTTP Error Status %s encountered for Service ID %s Instance %s under Job ID %s. No subsequent calls will be made.",
+					exception.getStatusCode().toString(), instance.getServiceId(), instance.getInstanceId(), instance.getJobId(),
+					instance.getNumberErrorResponses()), PiazzaLogger.WARNING);
+		}
+		// Remove this from the Instance Table
+		accessor.deleteAsyncServiceInstance(instance.getJobId());
+		// Send the Kafka Message for successful Cancellation status
+		StatusUpdate statusUpdate = new StatusUpdate(StatusUpdate.STATUS_CANCELLED);
+		try {
+			producer.send(JobMessageFactory.getUpdateStatusMessage(instance.getJobId(), statusUpdate, SPACE));
```

Comment:
One-use variable `statusUpdate` could be eliminated.

`producer.send(JobMessageFactory.getUpdateStatusMessage(instance.getJobId(), new StatusUpdate(StatusUpdate.STATUS_CANCELLED), SPACE));`